IntegrationTests/Services/AWSS3IntegrationTests/S3FlexibleChecksumsTests.swift

```diff
@@ -0,0 +1,114 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AWSS3
+@testable import ClientRuntime
+
+final class S3FlexibleChecksumsTests: S3XCTestCase {
+
+    private enum UploadType {
+        case data(Data)
+        case stream(URL)
+    }
+
+    private func testPutGetObject(withChecksumAlgorithm algorithm: S3ClientTypes.ChecksumAlgorithm, objectNameSuffix: String, uploadType: UploadType) async throws {
+        let objectName = "test-\(objectNameSuffix)"
+        let byteStream: ByteStream
+
+        switch uploadType {
+        case .data(let data):
+            byteStream = ByteStream.data(data)
+        case .stream(let url):
+            let fileHandle = try XCTUnwrap(FileHandle(forReadingAtPath: url.relativePath))
+            byteStream = ByteStream.stream(FileStream(fileHandle: fileHandle))
+        }
+
+        let input = PutObjectInput(
+            body: byteStream,
+            bucket: bucketName,
+            checksumAlgorithm: algorithm,
+            key: objectName
+        )
+
+        let response = try await client.putObject(input: input)
+        let unwrappedResponse = try XCTUnwrap(response)
+
+        // Verify the checksum response based on the algorithm used.
+        XCTAssertNotNil(try XCTUnwrap(getChecksumResponse(from: unwrappedResponse, with: algorithm)))
+
+        let getInput = GetObjectInput(bucket: bucketName, checksumMode: S3ClientTypes.ChecksumMode.enabled, key: objectName)
+        let responseGet = try await client.getObject(input: getInput) // will error for normal payloads if checksum mismatch
+        XCTAssertNotNil(responseGet.body) // Ensure there's a body in the response.
+
+        // Additional step for stream: Validate stream and read data.
+        if case .stream = uploadType {
+            let streamingBody = try XCTUnwrap(responseGet.body)
+            if case .stream(let stream) = streamingBody {
+                XCTAssert(stream is ValidatingBufferedStream, "Expected ValidatingBufferedStream for streaming upload")
+                let data = try await streamingBody.readData() // will error if checksum mismatch
+                XCTAssertNotNil(data)
```

Comment:
I dont think that's necessary here, readData() since it is on a ValidatingBufferedStream will calculate a checksum and throw an error if it is not equal to the checksum of the stored object in S3. Therefore, this is already checking that the returned object is identical! Since if its not, we'd get different checksums

Reply:
Ah, I see, then there's no need - that makes sense 👍